bftengine/include/bftengine/CryptoManager.hpp

```diff
@@ -54,8 +54,18 @@ class CryptoManager : public IKeyExchanger, public IMultiSigKeyGenerator {
   // IMultiSigKeyGenerator methodsKEY_EX_LOG
   std::pair<std::string, std::string> generateMultisigKeyPair() override {
     // use default Cryptosystem for key generation
-    LOG_INFO(logger(), "Generating new multisig key pair");
-    return cryptoSystems_.at(0)->cryptosys_->generateNewKeyPair();
+    auto log = logger();
+    LOG_INFO(log, "Generating new multisig key pair");
+
+    try {
+      auto cryptoSystem = cryptoSystems_.at(0);
+      return cryptoSystem->cryptosys_->generateNewKeyPair();
+    }
+    catch(const std::out_of_range& e) {
+      LOG_FATAL(log, e.what());
+      LOG_FATAL(log, "cryptoSystems_.at(0) has failed!");
```

Comment:
Actually, you can combine those two in one log: LOG_FATAL(log, "cryptoSystems_.at(0) has failed! " << e.what);

Reply:
Thanks for pointing that out, I'll make the necessary changes.